pkg/controller/build/build_controller.go

```diff
@@ -48,8 +48,21 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 
 	pred := predicate.Funcs{
 		UpdateFunc: func(e event.UpdateEvent) bool {
+			buildRunFinalizer := false
+			// Check if the AnnotationBuildRunDeletion annotation is updated
+			if len(e.MetaOld.GetAnnotations()) > 0 && len(e.MetaNew.GetAnnotations()) > 0 {
```

Comment:
I think this logic is cumbersome, maybe something like the following is more simple?:

```
o := e.ObjectOld.(*build.Build)
n := e.ObjectNew.(*build.Build)

			
if !reflect.DeepEqual(o.ObjectMeta.Annotations, n.ObjectMeta.Annotations) {
  return n.ObjectMeta.Annotations[build.AnnotationBuildRunDeletion
}
return false
```

Reply:
The conditions are what I summarized, it covers all cases.

First of all, not just `buildRunDeletion` annotation in the annotations, there is a kube LastApplied annotation is changed everytime when the resource is changed.
At that time, the old and new annotations are different, then the `return n.ObjectMeta.Annotations[build.AnnotationBuildRunDeletion` is always true. But maybe the `BuildRunDeletion` is not changed at this time

Reply:
@zhangtbj if I get u what u re saying, then you just need to do
```
			if !reflect.DeepEqual(oldAnnot, newAnnot) {
				if oldAnnot[build.AnnotationBuildRunDeletion] != newAnnot[build.AnnotationBuildRunDeletion] {
					return true
				}
			}
			return false
```
this just simplifies your code and make it readable, can you consider this one more time?

Reply:
yes, it looks better. Thanks Enrique, changed to:
```
			if !reflect.DeepEqual(oldAnnot, newAnnot) {
				if oldAnnot[build.AnnotationBuildRunDeletion] != newAnnot[build.AnnotationBuildRunDeletion] {
					return buildRunFinalizer
				}
			}
```

And I have confrim, because we don't need to add annotation, so we don't need to check if Annotations is nil.